### PR TITLE
Tunl64

### PR DESCRIPTION
**Pull request by ywc689**

### CODE REVIEW

File: src/ipvs/ip_vs_xmit.c [hunk 1]
```diff
@@ -2255,16 +2246,112 @@ static int __dp_vs_xmit_tunnel6(struct dp_vs_proto *proto,
...
+    rte_pktmbuf_free(mbuf);
+    return err;
+}
+
 int dp_vs_xmit_tunnel(struct dp_vs_proto *proto,
                       struct dp_vs_conn *conn,
                       struct rte_mbuf *mbuf)
 {
-    int af = conn->af;
+    int iaf = conn->af;
```
Comment (mscbg) [inline src/ipvs/ip_vs_xmit.c]: It is better use tuplehash_in.af

File: src/ipvs/ip_vs_xmit.c [hunk 2]
```diff
@@ -2255,16 +2246,112 @@ static int __dp_vs_xmit_tunnel6(struct dp_vs_proto *proto,
...
+        return __dp_vs_xmit_tunnel4(proto, conn, mbuf);
+
+    /* ip6-ip6 tunnel */
+    if (AF_INET6 == oaf)
+        return __dp_vs_xmit_tunnel6(proto, conn, mbuf);
 
-    return af == AF_INET ? __dp_vs_xmit_tunnel4(proto, conn, mbuf)
-        : __dp_vs_xmit_tunnel6(proto, conn, mbuf);
+    /* ip6-over-ip4 tunnel */
+    return __dp_vs_xmit_tunnel_6o4(proto, conn, mbuf);
```
Comment (mscbg) [inline src/ipvs/ip_vs_xmit.c]: It is better to judge in and out af, then choose tunnel4/tunnel6/tunnel64. I think we should not decide only tunnel4，tunnle6，tunnle64 can be work in dp_vs_xmit_tunnel.

File: src/ipv6/route6_lpm.c [hunk 3]
```diff
@@ -219,9 +219,9 @@ static struct route6 *rt6_lpm_input(const struct rte_mbuf *mbuf, struct flow6 *f
     /* FIXME: search hash list for detailed match ? */
     if (rt6->rt6_dev && fl6->fl6_oif && rt6->rt6_dev->id != fl6->fl6_oif->id)
         goto miss;
-    if (!ipv6_addr_any(&rt6->rt6_src.addr) && !ipv6_addr_any(&fl6->fl6_saddr)
-            && !ipv6_addr_equal(&rt6->rt6_src.addr, &fl6->fl6_saddr))
-        goto miss;
+    // if (!ipv6_addr_any(&rt6->rt6_src.addr) && !ipv6_addr_any(&fl6->fl6_saddr)
+    //         && !ipv6_addr_equal(&rt6->rt6_src.addr, &fl6->fl6_saddr))
+    //    goto miss;
```
Comment (mscbg) [inline src/ipv6/route6_lpm.c]: delete useless code

File: src/ipv6/route6_lpm.c [hunk 4]
```diff
@@ -243,9 +243,9 @@ static struct route6 *rt6_lpm_output(const struct rte_mbuf *mbuf, struct flow6 *
     /* FIXME: search hash list for detailed match ? */
     if (rt6->rt6_dev && fl6->fl6_oif && rt6->rt6_dev->id != fl6->fl6_oif->id)
         goto miss;
-    if (!ipv6_addr_any(&rt6->rt6_src.addr) && !ipv6_addr_any(&fl6->fl6_saddr)
-                && !ipv6_addr_equal(&rt6->rt6_src.addr, &fl6->fl6_saddr))
-        goto miss;
+    // if (!ipv6_addr_any(&rt6->rt6_src.addr) && !ipv6_addr_any(&fl6->fl6_saddr)
+    //             && !ipv6_addr_equal(&rt6->rt6_src.addr, &fl6->fl6_saddr))
+    //    goto miss;
```
Comment (mscbg) [inline src/ipv6/route6_lpm.c]: delete useless code

File: src/ipv6/route6_hlist.c [hunk 5]
```diff
@@ -280,9 +280,9 @@ rt6_hlist_flow_match(const struct route6 *rt6, const struct flow6 *fl6)
...
         return false;
 
-    if ((!ipv6_addr_any(&rt6->rt6_src.addr)) && (ipv6_addr_equal(&rt6->rt6_src.addr,
-                    &fl6->fl6_saddr)) != true)
-        return false;
+    // if ((!ipv6_addr_any(&rt6->rt6_src.addr)) && (!ipv6_addr_any(&fl6->fl6_saddr)) &&
+    //         (ipv6_addr_equal(&rt6->rt6_src.addr, &fl6->fl6_saddr)) != true)
+    //     return false;
 
     /* anything else to check ? */
```
Comment (mscbg) [inline src/ipv6/route6_hlist.c]: delete useless code

File: src/ipvs/ip_vs_xmit.c (hunk 2 above)
Comment (ywc689) [reply src/ipvs/ip_vs_xmit.c]: For IP protocol, there exists 3 types of tunnel only:
iaf=IPv4, oaf=IPv4 --> ip-ip tunnel
iaf=IPv6, oaf=IPv6 --> ip6-ip6 tunnel
iaf=IPv6, oaf=IPv4 --> sit tunnel
It does really not need to judge both iaf and oaf here.

**Pull request merged**